app/controllers/miq_request_controller.rb

```diff
@@ -400,6 +400,16 @@ def states_for_checkboxes_i18n
     end
   end
 
+  def user_options(params)
+    {
+      :reason_text    => params["reasonText"] || prov_set_default_options[:reason_text],
+      :applied_states => params["approvalStateCheckboxes"] || prov_set_default_options[:applied_states],
+      :type_choice    => params["types"] || prov_set_default_options[:type_choice],
+      :user_choice    => params["selectedUser"] || prov_set_default_options[:user_choice],
+      :time_period    => params["selectedPeriod"] || prov_set_default_options[:time_period],
+    }
```

Comment:
`prov_set_default_options` getting called here over and over is kind of inefficient (yeah I know we said to undo it haha).  I noticed that the keys match between the user and default options, so I think this might be cleaner:

```suggestion
    opts = {
      :reason_text    => params["reasonText"]
      :applied_states => params["approvalStateCheckboxes"]
      :type_choice    => params["types"]
      :user_choice    => params["selectedUser"]
      :time_period    => params["selectedPeriod"]
    }.compact
    prov_set_default_options.merge(opts)
```